crates/core/tedge_actors/src/test_helpers.rs

```diff
@@ -14,7 +15,122 @@ use crate::SimpleMessageBoxBuilder;
+use std::collections::HashSet;
 use std::fmt::Debug;
+use std::hash::Hash;
+use std::time::Duration;
+
+/// Check that a list of messages has been received in the given order
+///
+/// ```rust
+/// use crate::tedge_actors::{Builder, NoMessage, RuntimeError, ServiceConsumer, SimpleMessageBox, SimpleMessageBoxBuilder, test_helpers};
+///
+/// #[derive(Debug,Eq,PartialEq,Hash)]
+/// enum MyMessage {
+///    Foo(u32),
+///    Bar(u32),
+/// }
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(),RuntimeError> {
+///
+/// use std::time::Duration;
+/// let mut receiver_builder = SimpleMessageBoxBuilder::new("Recv", 16);
+/// let sender_builder = SimpleMessageBoxBuilder::new("Send", 16).with_connection(&mut receiver_builder);
+/// let mut sender = sender_builder.build();
+/// let mut receiver: SimpleMessageBox<MyMessage,NoMessage> = receiver_builder.build();
+///
+/// sender.send(MyMessage::Foo(1)).await?;
+/// sender.send(MyMessage::Bar(2)).await?;
+/// sender.send(MyMessage::Foo(3)).await?;
+///
+/// test_helpers::assert_received(&mut receiver, Duration::from_millis(100), [
+///     MyMessage::Foo(1),
+///     MyMessage::Bar(2),
+///     MyMessage::Foo(3),
+/// ]);
+///
+/// # Ok(())
+/// # }
+///
+/// ```
+pub async fn assert_received<T, U>(
+    messages: &mut impl ReceiveMessages<U>,
+    timeout: Duration,
+    expected: T,
+) where
+    T: IntoIterator,
+    U: From<T::Item>,
+    U: Debug + Eq,
+{
+    for expected_msg in expected.into_iter().map(|msg| msg.into()) {
+        let actual_msg = tokio::time::timeout(timeout, messages.recv()).await;
+        assert_eq!(actual_msg, Ok(Some(expected_msg)));
+    }
+}
+
+/// Pull the received `messages` until all the `expected` messages have been received or timeout reached.
+///
+/// A message is expected if containing one of the `expected` strings.
+/// Returns early on `timeout` while waiting for the next message.
+///
+/// ```rust
+/// use crate::tedge_actors::{Builder, NoMessage, RuntimeError, ServiceConsumer, SimpleMessageBox, SimpleMessageBoxBuilder, test_helpers};
+///
+/// #[derive(Debug,Eq,PartialEq,Hash)]
+/// enum MyMessage {
+///    Foo(u32),
+///    Bar(u32),
+/// }
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(),RuntimeError> {
+///
+/// use std::time::Duration;
+/// let mut receiver_builder = SimpleMessageBoxBuilder::new("Recv", 16);
+/// let sender_builder = SimpleMessageBoxBuilder::new("Send", 16).with_connection(&mut receiver_builder);
+/// let mut sender = sender_builder.build();
+/// let mut receiver: SimpleMessageBox<MyMessage,NoMessage> = receiver_builder.build();
+///
+/// sender.send(MyMessage::Foo(1)).await?;
+/// sender.send(MyMessage::Bar(2)).await?;
+/// sender.send(MyMessage::Foo(3)).await?;
+///
+/// test_helpers::assert_received_all_expected(&mut receiver, Duration::from_millis(100), [
+///     MyMessage::Foo(3),
+///     MyMessage::Bar(2),
+/// ]);
+///
+/// # Ok(())
+/// # }
+///
+/// ```
+pub async fn assert_received_all_expected<T, U>(
```

Comment:
```suggestion
pub async fn assert_received_unordered<T, U>(
```
We could add an `assert_received_ordered` later on, that will check if all the expected messages were received in that order itself. It's different from the above `assert_received` method in the sense that it allows `messages` to have more messages than what's provided in `expected`.